data/src/test/java/org/apache/iceberg/data/DataTest.java

```diff
@@ -160,6 +169,8 @@ public void testMixedTypes() throws IOException {
         )))
     );
 
+    Schema schema = new Schema(TypeUtil.assignFreshIds(structType, this::assignNewId).asStructType().fields());
```

Comment:
You should be able to use `(new AtomicInteger(0))::incrementAndGet` instead. There is no need to create global state in the test cases.